### PR TITLE
[Access Log] Add EMIT_TIME command operator

### PR DESCRIPTION
**Pull request by IssaAbuKalbein**
Commit Message: Add EMIT_TIME command operator for access log.
Additional Description: Access log currently has only one command operator for timestamp (START_TIME), which is identical for all access logs of the same stream (In case we enable on-start and periodic access log). In this PR, I've added a command operator that presents the timestamp of emitting access log.
[command operator for the current time in access logging #28736](https://github.com/envoyproxy/envoy/issues/28736)
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

**Comment by soulxu** [comment]
@IssaAbuKalbein is this ready for review?

**Comment by IssaAbuKalbein** [comment]
> @IssaAbuKalbein is this ready for review?

Yes, it’s ready for review.

File: source/common/stream_info/stream_info_impl.h [hunk 1]
```diff
@@ -134,6 +134,8 @@ struct StreamInfoImpl : public StreamInfo {
 
   MonotonicTime startTimeMonotonic() const override { return start_time_monotonic_; }
 
+  SystemTime emitLogTime() const override { return time_source_.systemTime(); }
```
Comment (RyanTheOptimist) [inline source/common/stream_info/stream_info_impl.h:137]: Hm. This is a bit surprising. This method simply returns the current time. It doesn't return the time the log was emitted, so much as it returns the current time and it's called by the logger so at *that* time it returns the time the log was emitted. Perhaps we should expose time_source_ via a simple accessor instead? WDYT?

**Review by RyanTheOptimist — COMMENTED** [review]
/assign @zuercher

zuercher: since this is logging related, just want to double check this with you.

File: source/common/stream_info/stream_info_impl.h (hunk 1 above)
Comment (IssaAbuKalbein) [reply source/common/stream_info/stream_info_impl.h]: I agree, fixed thanks!

**Comment by wbpcode** [comment]
Hmmm, I think the %START_TIME% + %DURATION% should be same with the %EMIT_TIME%

**Review by zuercher — APPROVED** [review]
lgtm

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**